ios/testmanagerd/xcuitestrunner.go

```diff
@@ -504,42 +530,60 @@ func createTestConfigOnDevice(testSessionID uuid.UUID, info testInfo, houseArres
 	if err != nil {
 		return "", nskeyedarchiver.XCTestConfiguration{}, err
 	}
-	return xctestConfigPath, nskeyedarchiver.NewXCTestConfiguration(info.targetAppBundleName, testSessionID, info.targetAppBundleID, info.targetAppPath, testBundleURL, testsToRun, testsToSkip), nil
+	return xctestConfigPath, nskeyedarchiver.NewXCTestConfiguration(info.targetApp.targetAppBundleName, testSessionID, info.targetApp.targetAppBundleID, info.targetApp.targetAppPath, testBundleURL, testsToRun, testsToSkip), nil
 }
 
 func createTestConfig(info testInfo, testSessionID uuid.UUID, xctestConfigFileName string, testsToRun []string, testsToSkip []string) nskeyedarchiver.XCTestConfiguration {
 	// the default value for this generated by Xcode is the target name, and the same name is used for the '.xctest' bundle name per default
 	productModuleName := strings.ReplaceAll(xctestConfigFileName, ".xctest", "")
-	return nskeyedarchiver.NewXCTestConfiguration(productModuleName, testSessionID, info.targetAppBundleID, info.targetAppPath, "PlugIns/"+xctestConfigFileName, testsToRun, testsToSkip)
+	return nskeyedarchiver.NewXCTestConfiguration(productModuleName, testSessionID, info.targetApp.targetAppBundleID, info.targetApp.targetAppPath, "PlugIns/"+xctestConfigFileName, testsToRun, testsToSkip)
 }
 
 type testInfo struct {
-	testrunnerAppPath   string
-	testRunnerHomePath  string
+	testApp   testAppInfo
+	targetApp targetAppInfo // Optional
+}
+
+type testAppInfo struct {
+	testRunnerAppPath  string
+	testRunnerHomePath string
+}
+
+type targetAppInfo struct {
 	targetAppPath       string
 	targetAppBundleName string
 	targetAppBundleID   string
 }
 
-func getAppInfos(bundleID string, testRunnerBundleID string, apps []installationproxy.AppInfo) (testInfo, error) {
-	info := testInfo{}
+func getTestAppInfo(testRunnerBundleID string, apps []installationproxy.AppInfo) (testAppInfo, error) {
```

Comment:
why two different functions and two different structs for getting app infos?

you could collect all infos in a single struct, and thus with a single function
```
type appInfo struct {
	path       string
	bundleName string
	bundleID   string
	homePath   string
}

func getappInfo(bundleID string, apps []installationproxy.AppInfo) (appInfo, error) {
	for _, app := range apps {
		if app.CFBundleIdentifier == bundleID {
			info := appInfo{
				path:       app.Path,
				bundleName: app.CFBundleName,
				bundleID:   app.CFBundleIdentifier,
			}
			if home, ok := app.EnvironmentVariables["HOME"].(string); ok {
				info.homePath = home
			}
			return info, nil
		}
	}

	return appInfo{}, fmt.Errorf("Did not find test app for '%s' on device. Is it installed?", testRunnerBundleID)
}
```

Reply:
Thank you 🙇 Applied.